sympy/solvers/inequalities.py

```diff
@@ -475,7 +479,14 @@ def solve_univariate_inequality(expr, gen, relational=True, domain=S.Reals, cont
     else:
         e = expr.lhs - expr.rhs
         period = periodicity(e, gen)
-        if period is not None:
+        if period is S.Zero:
```

Comment:
Need to pull a test over (or write a new one) for this. If there period is 0, the function is constant: either 0 (so return domain), non-zero (return empty set), or unknown (handle below).